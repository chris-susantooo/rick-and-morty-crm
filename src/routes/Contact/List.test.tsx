import { Character, getCharacters } from 'rickmortyapi';
import { renderApp, screen, userEvent, waitFor } from 'test-utils';
import characters from './__mocks__/characters.json';

vi.mock('rickmortyapi');
const mockGetCharacters = vi.mocked(getCharacters);

const waitForRouteReady = async () => {
  expect(
    await screen.findByRole('heading', { name: 'Contact' })
  ).toBeInTheDocument();
};

const selectOption = async (triggerLabel: string, optionLabel: string) => {
  await userEvent.click(screen.getByRole('button', { name: triggerLabel }));
  await userEvent.click(screen.getByRole('menuitem', { name: optionLabel }));
};

describe('Contact list page', () => {
  beforeEach(() => {
    mockGetCharacters.mockResolvedValue({
      status: 200,
      data: { results: characters as Character[] },
      statusMessage: 'OK',
    });
  });

  afterEach(() => {
    vi.resetAllMocks();
  });

  it('should search as you type', async () => {
    renderApp({ route: '/contact' });
    await waitForRouteReady();

    expect(mockGetCharacters).toHaveBeenCalledWith({
      page: 1,
    });

    await userEvent.type(
      screen.getByPlaceholderText('Search Characters'),
      'rick'
    );

    await waitFor(() => {
      expect(mockGetCharacters).toHaveBeenCalledWith({
        name: 'rick',
        page: 1,
      });
    });

    expect(screen.getByDisplayValue('rick')).toBeInTheDocument();
  });

  it('should filter by status and gender', async () => {
    renderApp({ route: '/contact' });
    await waitForRouteReady();

    await selectOption('Status', 'Alive');
    expect(screen.getByRole('button', { name: 'Alive' })).toBeInTheDocument();

    await selectOption('Gender', 'Female');
    expect(screen.getByRole('button', { name: 'Female' })).toBeInTheDocument();
    await waitFor(() => {
      expect(mockGetCharacters).toHaveBeenCalledWith({
        status: 'alive',
        gender: 'female',
        page: 1,
      });
    });
  });

  it('should render reset filter button when there is an active filter', async () => {
    renderApp({ route: '/contact' });
    await waitForRouteReady();

    await selectOption('Status', 'Unknown');
    await selectOption('Gender', 'Genderless');
    await waitFor(() => {
      expect(mockGetCharacters).toHaveBeenCalledWith({
        status: 'unknown',
        gender: 'genderless',
        page: 1,
      });
    });

    await userEvent.click(screen.getByRole('button', { name: 'Reset' }));
    await waitFor(() => {
      expect(mockGetCharacters).toHaveBeenCalledWith({
        page: 1,
      });
    });
  });

  it('should render contact list items', async () => {
    const { asFragment } = renderApp({ route: '/contact' });
    await waitForRouteReady();

    const contacts = screen.getAllByRole('listitem');
    expect(contacts).toHaveLength(20);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render not found message on 404 error', async () => {
    mockGetCharacters.mockResolvedValue({
      status: 404,
      data: { results: [] },
      statusMessage: 'OK',
    });

    renderApp({ route: '/contact' });
    await waitForRouteReady();

    expect(await screen.findByText("No one's here ðŸ˜¿")).toBeInTheDocument();
  });

  it('should focus on search input when "/" is pressed', async () => {
    renderApp({ route: '/contact' });
    await waitForRouteReady();

    await userEvent.type(screen.getByRole('main'), '/');
    expect(screen.getByPlaceholderText('Search Characters')).toHaveFocus();
  });
});
