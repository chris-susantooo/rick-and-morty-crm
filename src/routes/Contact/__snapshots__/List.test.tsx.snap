// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Contact list page > should render contact list items 1`] = `
<DocumentFragment>
  <div
    class="flex h-screen flex-col overflow-y-auto lg:flex-row"
  >
    <aside
      class="h-full w-[16rem] shrink-0 flex-col border-e border-e-gray-100 px-6 fixed"
    >
      <nav
        class="mt-1 flex flex-col space-y-1"
      >
        <a
          class="my-3"
          href="/"
        >
          <h4
            class="scroll-m-20 text-xl font-semibold tracking-tight"
          >
            Rick and Morty
          </h4>
        </a>
        <a
          aria-current="page"
          class="active"
          href="/contact"
        >
          <button
            class="inline-flex items-center rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-10 px-4 py-2 w-full justify-start"
          >
            Contact
          </button>
        </a>
      </nav>
    </aside>
    <main
      class="flex h-[calc(100%-61px)] w-full lg:ms-64 lg:h-full flex-row"
    >
      <div
        class="flex h-full w-full flex-col pt-8 md:fixed md:w-[20rem] md:border-e md:border-e-gray-100 md:max-lg:h-[calc(100%-61px)]"
      >
        <h2
          class="scroll-m-20 pb-2 text-3xl font-semibold tracking-tight transition-colors first:mt-0 mx-6 mb-3"
        >
          Contact
        </h2>
        <form
          action="/contact"
          class="mx-6 mb-3"
          id="search-contact"
          method="get"
          role="search"
        >
          <div
            class="relative"
          >
            <label
              class="absolute top-[50%] translate-y-[-50%] ltr:left-3 rtl:right-3"
              for=":r13:"
            >
              <svg
                class="text-muted-foreground"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="11"
                  cy="11"
                  r="8"
                />
                <path
                  d="m21 21-4.3-4.3"
                />
              </svg>
            </label>
            <input
              class="flex h-10 w-full rounded-md border border-input bg-background px-3 py-[1.35rem] text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 ps-10 pe-10"
              id=":r13:"
              name="name"
              placeholder="Search Characters"
              value=""
            />
            <label
              class="absolute top-[50%] translate-y-[-50%] ltr:right-3 rtl:left-3"
              for=":r13:"
            >
              <kbd
                class="flex select-none rounded-[0.25rem] border border-gray-200 px-1.5"
              >
                <p
                  class="text-sm text-muted-foreground"
                >
                  /
                </p>
              </kbd>
            </label>
          </div>
          <div
            class="mt-3 flex justify-between"
          >
            <div
              class="flex gap-1"
            >
              <div
                class="relative"
              >
                <input
                  class="h-10 w-full rounded-md border border-input bg-background px-3 py-[1.35rem] text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 hidden"
                  id=":r14:"
                  name="status"
                  value=""
                />
              </div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 rounded-full bg-secondary text-secondary-foreground hover:bg-secondary/80 h-7 px-2 gap-1"
                data-state="closed"
                id="radix-:r15:"
                type="button"
              >
                <small
                  class="text-sm font-medium leading-none"
                >
                  Status
                </small>
                <svg
                  class="lucide lucide-chevron-down"
                  fill="none"
                  height="16"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
              <div
                class="relative"
              >
                <input
                  class="h-10 w-full rounded-md border border-input bg-background px-3 py-[1.35rem] text-sm ring-offset-background file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 hidden"
                  id=":r17:"
                  name="gender"
                  value=""
                />
              </div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex items-center justify-center text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 rounded-full bg-secondary text-secondary-foreground hover:bg-secondary/80 h-7 px-2 gap-1"
                data-state="closed"
                id="radix-:r18:"
                type="button"
              >
                <small
                  class="text-sm font-medium leading-none"
                >
                  Gender
                </small>
                <svg
                  class="lucide lucide-chevron-down"
                  fill="none"
                  height="16"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="m6 9 6 6 6-6"
                  />
                </svg>
              </button>
            </div>
          </div>
        </form>
        <div
          class="relative me-1 overflow-visible pe-5 ps-6 md:overflow-y-auto"
          dir="ltr"
          style="position: relative; --radix-scroll-area-corner-width: 0px; --radix-scroll-area-corner-height: 0px;"
        >
          <style>
            [data-radix-scroll-area-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-scroll-area-viewport]::-webkit-scrollbar{display:none}
          </style>
          <div
            class="h-full w-full rounded-[inherit]"
            data-radix-scroll-area-viewport=""
            style="overflow-x: hidden; overflow-y: hidden;"
          >
            <div
              style="min-width: 100%; display: table;"
            >
              <ul
                class="flex flex-col gap-1"
              >
                <a
                  class=""
                  href="/contact/1"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Rick Sanchez"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Rick Sanchez
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/2"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Morty Smith"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Morty Smith
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/3"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Summer Smith"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Summer Smith
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/4"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Beth Smith"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/4.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Beth Smith
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/5"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Jerry Smith"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/5.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Jerry Smith
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/6"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Abadango Cluster Princess"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/6.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Abadango Cluster Princess
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Alien
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/7"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Abradolf Lincler"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/7.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Abradolf Lincler
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/8"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Adjudicator Rick"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/8.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Adjudicator Rick
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/9"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Agency Director"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/9.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Agency Director
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/10"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Alan Rails"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/10.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Alan Rails
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/11"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Albert Einstein"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/11.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Albert Einstein
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/12"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Alexander"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/12.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Alexander
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/13"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Alien Googah"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/13.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Alien Googah
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Alien
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/14"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Alien Morty"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/14.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Alien Morty
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Alien
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/15"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Alien Rick"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/15.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Alien Rick
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Alien
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/16"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Amish Cyborg"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/16.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Amish Cyborg
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Alien
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/17"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Annie"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/17.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Annie
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/18"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Antenna Morty"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/18.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Antenna Morty
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/19"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Antenna Rick"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/19.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Antenna Rick
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
                <a
                  class=""
                  href="/contact/20"
                >
                  <li
                    class="flex cursor-pointer rounded-sm p-3 transition-colors hover:bg-gray-100"
                  >
                    <div
                      class="flex items-center gap-4"
                    >
                      <img
                        alt="Ants in my Eyes Johnson"
                        class="h-16 w-16 rounded-full bg-gray-200"
                        src="https://rickandmortyapi.com/api/character/avatar/20.jpeg"
                      />
                      <div
                        class="space-y-1"
                      >
                        <p
                          class="leading-7 [&:not(:first-child)]:mt-6"
                        >
                          Ants in my Eyes Johnson
                        </p>
                        <p
                          class="text-sm text-muted-foreground"
                        >
                          Human
                        </p>
                      </div>
                    </div>
                  </li>
                </a>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
